karyon3-core/src/main/java/com/netflix/karyon/AbstractKaryonSuite.java

```diff
@@ -0,0 +1,87 @@
+package com.netflix.karyon;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import com.google.inject.Module;
+import com.google.inject.Stage;
+import com.netflix.governator.GovernatorFeature;
+import com.netflix.governator.auto.ModuleListProvider;
+import com.netflix.governator.auto.PropertySource;
+
+public abstract class AbstractKaryonSuite implements KaryonSuite {
+    private KaryonBuilder builder;
+    
+    @Override
+    final public void configure(KaryonBuilder builder) throws Exception {
+        this.builder = builder;
+        configure();
+        this.builder = null;
+    }
+    
+    protected abstract void configure() throws Exception;
```

Comment:
I think if  we can flatten out all the ArchaiusKaryonSuite.Builder() methods, e.g, withConfigName, withApplicationOverrides, etc. here, then we can get rid of "using". I'll give the example in my mind of the final Karyon creation later.